### PR TITLE
organize loki a bit

### PR DESCRIPTION
**Pull request by kevinkreiser**
most of the changes here are in the loki benchmark. i reorganized it a bit so that i could most accurately measure changes to loki when we start working on fixing the problems with islands in #58 as well as #72 

specifically all i did was add placeholders in the interface to allow the user to request that we do these things. i havent actually implemented anything to do them.

### CODE REVIEW

File: valhalla/baldr/location.h [hunk 1]
```diff
@@ -99,13 +99,9 @@ struct Location {
...
   boost::optional<uint64_t> way_id_;
 
-  //TODO: fill these out in constructors and add getters and setters
-/*
-  //the spot where the feature is on the map
-  midgard::PointLL display_latlng_;
-  //id of the osm way that this location was on
-  std::uint64_t wayid_;
-*/
+  bool deisolate_; //should search for more edges if it has only found those on islands
```
Comment (kevinkreiser) [inline valhalla/baldr/location.h]: these are completely unused for now and subject to chance once we start actually working on #58 and #72

File: src/loki/search.cc [hunk 2]
```diff
@@ -380,30 +374,18 @@ struct ProjectPoint {
...
-      b->closest_point = point;
+      closest_point = point;
     }
   }
 
- private:
-  // To limit the size of the struct and optimize cache locality of
-  // the data used in the hot spot, we box the data not used in the
-  // hot spot.
-  struct Box {
```
Comment (kevinkreiser) [inline src/loki/search.cc]: in testing using `valhalla_benchmark_loki` the removal of this object actually sped up the code. for searches of a single location there was no change in speed. for searches where there more locations (like 200 or so) the code was actually faster with this removed

**Comment by dnesbitt61** [comment]
:shipit:

**Comment by kevinkreiser** [comment]
ok at this point ive finished refactoring loki::search back to the point where the logic to avoid finding only islands can be re-added. i havent actually added any of that logic back because i think im going to do it slightly differently but it shouldnt be as hard as it would have been

**Comment by dnesbitt61** [comment]
:shipit:

**Pull request merged**